<launch>
  <!-- ros_master_uri is the machine on which ROSCore is running -->
  <arg name="ros_master_uri" default="http://localhost:11311"/>
  <!-- ros_ip is the ip of the current machine -->
  <arg name="ros_ip" default="localhost"/>
  <arg name="num_drones" value="2"/>
  
  <!-- Set ROS Environment variables -->
  <env name="ROS_MASTER_URI" value="$(arg ros_master_uri)"/>
  <env name="ROS_IP" value="$(arg ros_ip)"/>
  <env name="ROS_HOSTNAME" value="$(arg ros_ip)"/>

  <!--  swarm bridge re-broadcasts the trajectories of all drones -->
  <include file="$(find swarm_bridge)/launch/bridge_udp.launch">
    <arg name="drone_id" value="999"/>
    <arg name="broadcast_ip" value="127.0.0.255"/>
  </include>

  <!-- Collision checker -->
  <node pkg="gestelt_bringup" type="swarm_collision_checker" 
    name="swarm_collision_checker" output="screen">
    <param name="~num_drones" value="$(arg num_drones)" />
    <param name="~collision_tolerance" value="0.6" />
    <param name="~check_collision_freq" value="10.0" />
  </node>

</launch>
